src/rhsm/connection.py

```
@@ -547,7 +550,23 @@ def _request(self, request_type, method, info=None):
                 id_cert = certificate.create_from_file(self.cert_file)
                 if not id_cert.is_valid():
                     raise ExpiredIdentityCertException()
-            raise
+                # we should retry here, its not an identity cert issue
+                retry = True
+                time.sleep(1)
+
+        if retry:
+            for num in range(ssl_retries):
+                log.warn("Retrying connection request")
+                try:
+                    conn.request(request_type, handler, body=body, headers=headers)
+                    break
+                except Exception:
+                    log.warn("Retry SSL exception")
+                    if num == ssl_retries-1:
```

We should probably throw the exception that we caught.